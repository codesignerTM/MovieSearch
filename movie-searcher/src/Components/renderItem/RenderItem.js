import React from "react";
import "./RenderItem.scss";

const RenderItem = props => {
  return (
    <>
      <li>
        <button
          className="render-item"
          onClick={props.onClick}
          data-testid="render-item-button"
        >
          {props.movieName}
        </button>
      </li>
    </>
  );
};

export default RenderItem;
