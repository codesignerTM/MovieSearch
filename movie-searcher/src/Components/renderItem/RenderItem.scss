.render-item {
  width: 300px;
  margin: 3px auto;
  padding: 5px;
  background-color: #ccc;
  color: #fff;
  border-radius: 3px;
}

.render-item:hover {
  background-color: rgb(174, 181, 238);
}
