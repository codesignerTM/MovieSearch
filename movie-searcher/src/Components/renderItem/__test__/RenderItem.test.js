import React from "react";
import ReactDOM from "react-dom";
import RenderItem from "../RenderItem";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crash", () => {
  const div = document.createElement("div");
  ReactDOM.render(<RenderItem />, div);
});

it("renders correctly", () => {
  const { getByTestId } = render(
    <RenderItem onClick={() => {}} movieName="hello" />
  );
  expect(getByTestId("render-item-button")).toHaveTextContent("hello");
});

it("renders correctly", () => {
  const { getByTestId } = render(
    <RenderItem onClick={() => {}} movieName="save" />
  );
  expect(getByTestId("render-item-button")).toHaveTextContent("save");
});

it("matches snapshot 1", () => {
  const tree = renderer
    .create(<RenderItem onClick={() => {}} movieName="save" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot 2", () => {
  const tree = renderer
    .create(<RenderItem onClick={() => {}} movieName="matrix" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
