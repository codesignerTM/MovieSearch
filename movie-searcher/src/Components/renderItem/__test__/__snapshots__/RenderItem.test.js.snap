// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li>
  <button
    className="render-item"
    data-testid="render-item-button"
    onClick={[Function]}
  >
    save
  </button>
</li>
`;

exports[`matches snapshot 1 1`] = `
<li>
  <button
    className="render-item"
    data-testid="render-item-button"
    onClick={[Function]}
  >
    save
  </button>
</li>
`;

exports[`matches snapshot 2 1`] = `
<li>
  <button
    className="render-item"
    data-testid="render-item-button"
    onClick={[Function]}
  >
    matrix
  </button>
</li>
`;
